syntax = "proto3";

package ai.stigmer.commons.apiresource;

import "ai/stigmer/commons/apiresource/enum.proto";
import "buf/validate/validate.proto";

// ApiResourceMetadata contains standard metadata for all API resources.
message ApiResourceMetadata {
  // Human-readable name of the resource.
  string name = 1;

  // URL-friendly identifier, unique within scope.
  string slug = 2;

  // System-generated unique identifier.
  string id = 3;

  // Key-value labels for organization and filtering.
  map<string, string> labels = 4;

  // Key-value annotations for additional metadata.
  map<string, string> annotations = 5;

  // Tags for categorization.
  repeated string tags = 6;

  // Version information for the resource.
  ApiResourceMetadataVersion version = 7;
}

// ApiResourceMetadataVersion contains version tracking information.
message ApiResourceMetadataVersion {
  // Unique identifier for this version.
  string id = 1;

  // Message describing what changed in this version.
  string message = 2;

  // Reference to the previous version ID.
  string previous_version_id = 3;
}
